internal/extproc/translator/gemini_helper.go

```diff
@@ -20,30 +28,488 @@ const (
 	HTTPHeaderKeyContentLength = "Content-Length"
 )
 
+// -------------------------------------------------------------
+// Request Conversion Helper for OpenAI to GCP Gemini Translator
+// -------------------------------------------------------------
+
+// toGeminiContents converts OpenAI messages to Gemini Contents and SystemInstruction
+func toGeminiContents(messages []openai.ChatCompletionMessageParamUnion) ([]genai.Content, *genai.Content, error) {
+	var gcpContents []genai.Content
+	var systemInstruction *genai.Content
+	knownToolCalls := make(map[string]string)
+	var gcpParts []*genai.Part
+
+	for _, msgUnion := range messages {
+		switch msgUnion.Type {
+		case openai.ChatMessageRoleDeveloper:
+			msg := msgUnion.Value.(openai.ChatCompletionDeveloperMessageParam)
+			inst, err := fromDeveloperMsg(msg)
+			if err != nil {
+				return nil, nil, fmt.Errorf("error converting developer message: %w", err)
+			}
+			if len(inst) != 0 {
+				if systemInstruction == nil {
+					systemInstruction = &genai.Content{Role: genai.RoleUser}
```

Comment:
maybe put code comment on why this user insertion is needed? Let's make sure that any non trivial logic like this should have a code comment for future contributors.